### PR TITLE
bon eval

### PR DESCRIPTION
**Pull request by yuchenlin**

### CODE REVIEW

**Comment by natolambert** [comment]
Will give a more thorough review once the data is moved.
You can use the function `save_to_hub` to easily do this.
Would be good to just add the data conversion scripts in the source `rewardbench/` repo, then call them in the script so it's all automatic end to end.

**Review by natolambert — CHANGES_REQUESTED** [review]
This is getting closer. High level comments also shared on slack.
1. We still have more jsons to move to huggingface
2. Make all of this happen in the existing `scripts/run_bon.py`. 
3. To do so, most of the functions can be in `rewardbench/bon_utils.py`
4. We can have duplicated scripts in analysis in case there's an error and someone wants to re-run alpacaeval conversion
5. Code cleaning, adding license, removing unnecessary things

Mostly, I want to make this first-party supported in the repo, not a separate folder.

File: bon_eval/bon_utils.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
+import logging
+import sys
+from pathlib import Path
+from typing import Any, Callable, Literal, Optional, Sequence, Union
+
+import fire
+import pandas as pd
+
+from alpaca_eval import analyze, annotators, constants, decoders, metrics, utils
```
[the remark this answers is not in the document]
Comment (yuchenlin) [reply bon_eval/bon_utils.py:9]: In the new commits, I've removed unnecessary imports, the kept ones are all needed in this script.

File: bon_eval/bon_utils.py [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    else:
+        utils.print_leaderboard(
+            df_leaderboard,
+            leaderboard_mode_to_print,
+            current_name=name,
+            cols_to_print=[sort_by, "win_rate", "standard_error", "n_total", "avg_length"],
+        )
+
+ 
+if __name__ == "__main__":
```
[the remark this answers is not in the document]
Comment (yuchenlin) [reply bon_eval/bon_utils.py:256]: it was used for testing and showing the usage. now it's not useful and I've removed it

File: bon_eval/bon_utils.py [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+from typing import Any, Callable, Literal, Optional, Sequence, Union
+
+import fire
+import pandas as pd
+
+from alpaca_eval import analyze, annotators, constants, decoders, metrics, utils
+from alpaca_eval.types import AnyData, AnyLoadableDF, AnyPath
+ 
+
+def evaluate(
```
[the remark this answers is not in the document]
Comment (yuchenlin) [reply bon_eval/bon_utils.py:13]: added a comment: # modified from https://github.com/tatsu-lab/alpaca_eval/blob/c4a4ca716b4cab46701af759c244cb9d05772e15/src/alpaca_eval/main.py#L17C5-L17C5

**Pull request closed without merging**